### ISSUE TITLE
Moving Window doesn't Work

### ISSUE THREAD

**Issue by anlumo** [horriblename/hyprgrass#146]
According to #70, the following configuration should allow window dragging:

```
bindm = ,longpress:3, movewindow
```

When I add this, the longpress causes the window to float temporarily, but I can't move it by moving my fingers. Moving using a mouse works, but that kinda defeats the point. Once I remove the fingers, the window moves back to where it was in the layout (since it was dropped at the initial location).

Maybe something in hyprland changed since that PR?

**Comment by horriblename** [comment]
does this work?

```
hyprgrass-bindm =, longpress:3, movewindow
```

use of normal bindm is deprecated and unsupported

**Comment by anlumo** [comment]
Thank you for the response. Unfortunately, this shows the same behavior.

I'm on the hyprland-git AUR, btw., because the regular package doesn't allow installing plugins at the moment.

**Comment by anlumo** [comment]
In my attempts to set up my tablet, I also noticed that the hyprland option `resize_on_border` only allows resizing with a mouse/trackpad, not touch or pen. It looks like there could be an event filter on the hyprland side that ignores these kinds of input devices for window manipulation.

When initiating a window move operation using a mouse, the touchscreen behaves weirdly. When I tap somewhere the window is centered on that location, but I can't drag it using the touchscreen (the window just doesn't move).

**Comment by horriblename** [comment]
resize_on_border has never worked on touch.

plugins seem to be broken at the moment but could you tell me what commit of hyprland you're on?

**Comment by anlumo** [comment]
the version is `0.42.0.r1.83a334f9-1`

**Comment by anlumo** [comment]
Ok, that version of hyprland was competely broken, it didn't even compile one of its files needed for plugins properly (the joy of having to use a -git package). `0.42.0.r46.c86db7bb-1` can load hyprgrass again, but the window drag issue is unchanged from my initial description.

**Comment by horriblename** [comment]
just as a sanity check, you're on a touch screen right? hyprgrass doesn't work on touchpads (I have been getting this a lot)

**Comment by anlumo** [comment]
Yes, I'm running it on a Minisforum V3, which has both a touchscreen and a touchpad. My tests were only on the touchscreen.

**Comment by horriblename** [comment]
still can't reproduce on hyprland latest : (

can you try disabling `cursor:hide_on_touch` and send me a recording?

plus your hyprland config

**Comment by anlumo** [comment]
[Config file](https://gist.github.com/anlumo/3b5b5850d9b482d3a4b77abe96fdc49f)

Video:

https://github.com/user-attachments/assets/86c0d44d-914f-4344-8ce6-d69c826353ba

**Comment by horriblename** [comment]
so `cursor:no_warps` breaks movewindow for some ungodly reason,

still not sure why, but I'll keep digging

P.S. `experimental:send_cancel` has been broken for a while and you should disable it

**Comment by horriblename** [comment]
after reading the wiki on `no_warps` I have no clue what it does. Can you tell me what it's supposed to do?

**Comment by anlumo** [comment]
Yes that fixed it, thank you!

Should I keep the ticket open to track the `no_warps` incompatibility?

> after reading the wiki on `no_warps` I have no clue what it does. Can you tell me what it's supposed to do?

When I press Meta+Arrow (with my config), the focus jumps to a different window. With `no_warps=false`, the mouse cursor stays where it was. With `no_warps=true`, the mouse cursor jumps to the center of the newly focused window.

**Comment by horriblename** [comment]
> Should I keep the ticket open to track the no_warps incompatibility?

yes keep it open please

**Comment by sr-tream** [comment]
With multi-monitor setup and `no_warps = true` any gestures applied to display with mouse, as workaround can be used this config:
```
    no_warps = false
    warp_back_after_non_mouse_input = true
```

They prevent to move mouse to monitor with touchscreen, but no solution for fully-featured `no_warps = true`
